tracer/test/Datadog.Trace.DuckTyping.Tests/GetAssemblyTests.cs

```diff
@@ -0,0 +1,55 @@
+// <copyright file="GetAssemblyTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Xunit;
+
+namespace Datadog.Trace.DuckTyping.Tests
+{
+    [Collection(nameof(GetAssemblyTestsCollection))]
+    public class GetAssemblyTests
+    {
+        [Fact]
+        public void GetAssemblyTest()
+        {
+            var asmDuckTypes = 0;
+            var lstExceptions = new List<Exception>();
+            var assemblies = System.AppDomain.CurrentDomain.GetAssemblies();
+            foreach (var assembly in assemblies)
+            {
+                if (assembly.FullName!.StartsWith(DuckTypeConstants.DuckTypeAssemblyPrefix) ||
+                    assembly.FullName!.StartsWith(DuckTypeConstants.DuckTypeGenericTypeAssemblyPrefix) ||
+                    assembly.FullName!.StartsWith(DuckTypeConstants.DuckTypeNotVisibleAssemblyPrefix))
+                {
+                    asmDuckTypes++;
+
+                    try
+                    {
+                        assembly.GetTypes();
+                    }
+                    catch (ReflectionTypeLoadException ex)
+                    {
+                        lstExceptions.AddRange(ex.LoaderExceptions);
+                    }
+                }
+            }
+
+            if (lstExceptions.Count > 0)
+            {
+                throw new AggregateException(lstExceptions.ToArray());
+            }
+
+#if NETFRAMEWORK
```

Comment:
nit: might be nice to add a comment that this number is expected to change if you add another test to the duck types assembly

Reply:
Comment added in https://github.com/DataDog/dd-trace-dotnet/pull/2716/commits/7a8a8942d10547d8c3d564471e01f9631f2c635d